src/arrow_writer.rs

```diff
@@ -0,0 +1,318 @@
+use std::io::{Seek, Write};
+
+use arrow::{
+    array::RecordBatch,
+    datatypes::{DataType as ArrowDataType, SchemaRef},
+};
+use prost::Message;
+use snafu::{ensure, ResultExt};
+
+use crate::{
+    error::{IoSnafu, Result, UnexpectedSnafu},
+    proto,
+    writer::{
+        column::EstimateMemory,
+        stripe::{StripeInformation, StripeWriter},
+    },
+};
+
+/// Construct an [`ArrowWriter`] to encode [`RecordBatch`]es into a single
+/// ORC file.
+pub struct ArrowWriterBuilder<W> {
+    writer: W,
+    schema: SchemaRef,
+    batch_size: usize,
+    stripe_byte_size: usize,
+}
+
+impl<W: Write + Seek> ArrowWriterBuilder<W> {
```

Comment:
Do we really need `Seek` to write orc file?

Reply:
Let me experiment with removing the need for seek :thinking:

Reply:
Thanks for picking up on this, I guess I reached for Seek initially since was naturally thinking of native fs files first. Removed need for Seek :+1: